idx/memory/memory.go

```diff
@@ -64,6 +64,30 @@ type TagIDs map[idx.MetricID]struct{} // set of ids
 type TagValue map[string]TagIDs       // value -> set of ids
 type TagIndex map[string]TagValue     // key -> list of values
 
+func (t *TagIndex) AddTagId(name, value string, id idx.MetricID) {
+	ti := *t
+	if _, ok := ti[name]; !ok {
+		ti[name] = make(TagValue)
+	}
+	if _, ok := ti[name][value]; !ok {
+		ti[name][value] = make(TagIDs)
+	}
+	ti[name][value][id] = struct{}{}
+}
+
+func (t *TagIndex) DelTagId(name, value string, id idx.MetricID) {
```

Comment:
this function (and the Add one) don't need to be exported I think?

Reply:
:+1: https://github.com/grafana/metrictank/pull/759/commits/107aef7299ec3a230cf1e3fc2e085547e352a345